tuned/exports/unix_socket_exporter.py

```diff
@@ -0,0 +1,246 @@
+
+	def register_signal_path(self, path):
+		self._socket_signal_paths.append(path)
+
+	def _construct_socket_object(self):
+		if self._socket_path:
+			if os.path.exists(self._socket_path):
+				os.unlink(self._socket_path)
+			self._socket_object = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+			self._socket_object.bind(self._socket_path)
+			self._socket_object.listen(self._connections_backlog)
+			os.chown(self._socket_path, self._ownership[0], self._ownership[1])
+			if self._permissions:
+				os.chmod(self._socket_path, self._permissions)
+
+	def start(self):
+		if self.running():
+			return
+
+		self.stop()
+		self._construct_socket_object()
+
+	def stop(self):
+		if self._socket_object:
+			self._socket_object.close()
+
+	def _send_data(self, s, data):
+		log.debug("Sending socket data: %s)" % data)
+		try:
+			s.send(json.dumps(data).encode("utf-8"))
+		except Exception as e:
+			log.warning("Failed to send data '%s' type '%s': %s" % (data, e))
+
+	def _create_response(self, data, id, error=False):
+		res = {
+			"jsonrpc": "2.0",
+			"id": id
+		}
+		if error:
+			res["error"] = data
+		else:
+			res["result"] = data
+		return res
+
+	def _create_error_responce(self, code, message, id=None, data=None):
+		return self._create_response({
+			"code": code,
+			"message": message,
+			"data": data,
+		}, error=True, id=id)
+
+	def _create_result_response(self, result, id):
+		return self._create_response(result, id)
+
+	def _check_id(self, data):
+		if data.get("id"):
+			return data
+		return None
+
+	def _process_request(self, req):
+		if type(req) != dict or req.get("jsonrpc") != "2.0" or not req.get("method"):
+			return self._create_error_responce(-32600, "Invalid Request")
+		id = req.get("id")
+		ret = None
+		if req["method"] not in self._unix_socket_methods:
+			return self._check_id(self._create_error_responce(-32601, "Method not found", id))
+		try:
+			if not req.get("params"):
+				ret = self._unix_socket_methods[req["method"]]()
+			elif type(req["params"]) in (list, tuple):
+				ret = self._unix_socket_methods[req["method"]](*req["params"])
+			elif type(req["params"]) == dict:
+				ret = self._unix_socket_methods[req["method"]](**req["params"])
+			else:
+				return self._check_id(self._create_error_responce(-32600, "Invalid Request", id))
+		except TypeError as e:
+			return self._check_id(self._create_error_responce(-32602, "Invalid params", id, str(e)))
+		except Exception as e:
+			return self._check_id(self._create_error_responce(1, "Error", id, str(e)))
+		return self._check_id(self._create_result_response(ret, id))
+
+	def period_check(self):
+		"""
+		Periodically checks socket object for new calls. This allows to function without special thread.
+		Interface is according JSON-RPC 2.0 Specification (see https://www.jsonrpc.org/specification)
+		
+		Example calls:
+		
+		printf '[{"jsonrpc": "2.0", "method": "active_profile", "id": 1}, 1]' | sudo nc -U /run/tuned/tuned.sock
```

Comment:
I believe we can provide a better example combining the calls.  This one returns "Invalid request" and might confuse people.

Perhaps this one?

```
'[{"jsonrpc": "2.0", "method": "active_profile", "id": 1}, {"jsonrpc": "2.0", "method": "profiles", "id": 2}]'
```

Reply:
I wanted to provide example when call fails, but you're right it'd be unnecessarily confusing